dali/kernels/test/tensor_test_utils.h

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2018-2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_TEST_TENSOR_TEST_UTILS_H_
+#define DALI_KERNELS_TEST_TENSOR_TEST_UTILS_H_
+
+#include <gtest/gtest.h>
+#include <functional>
+#include <cmath>
+#include <utility>
+#include <random>
+#include "dali/kernels/tensor_view.h"
+#include "dali/kernels/tensor_shape_str.h"
+#include "dali/kernels/backend_tags.h"
+#include "dali/kernels/util.h"
+
+namespace dali {
+namespace kernels {
+
+// COMPARISON
+
+template <int dim1, int dim2>
+void CheckEqual(
+    const TensorListShape<dim1> &s1,
+    const TensorListShape<dim2> &s2,
+    int &max_errors) {
+  ASSERT_EQ(s1.sample_dim(), s2.sample_dim()) << "Tensor lists must have equal sample dimension";
+  ASSERT_EQ(s1.num_samples(), s2.num_samples()) << "Tensor lists must have same number of samples";
+  int n = s1.num_samples();
+  int errors = 0;
+  for (int i = 0; i < n; i++) {
+    if (s1.tensor_shape_span(i) != s2.tensor_shape_span(i)) {
+      if (++errors < max_errors) {
+        EXPECT_EQ(s1.tensor_shape_span(i), s2.tensor_shape_span(i))
+            << "Samples #" << (--max_errors, i)
+            << " have different shapes " << s1[i] << " vs " << s2[i];
+      }
+    }
+  }
+
+  if (errors) {
+    if (errors > max_errors) {
+      max_errors = 0;
+      FAIL() << " (" << (errors - max_errors) << " more)";
+    } else {
+      max_errors -= errors;
+      FAIL();
+    }
+  }
+}
+
+template <int dim1, int dim2>
+void CheckEqual(const TensorListShape<dim1> &s1, const TensorListShape<dim2> &s2) {
+  int max_errors = 100;
```

Comment:
Why 100?

Reply:
Also this is value that is getting repeated in other functions, maybe put it in one place?

Reply:
Done.